### ISSUE TITLE
Render twice using hook component

### ISSUE THREAD

**Issue by musicq** [facebook/react-devtools#1333]
Hook components will render twice when open react devtool.

```jsx
import React, { useState } from 'react';

function App() {
  const [count, setCount] = useState(0);
  const incrementCount = () => setCount(count + 1);

  console.log('render');

  return <button onClick={incrementCount}>{count}</button>;
}

export default App;
```

Click the button, the console will print twice 'render' message. If you close React devtool, it'll print only once.

This seems a bug of react devtool.

**Comment by eawww** [comment]
@musicq I'm having a similar problem. How do you attribute it to react-devtools?

**Comment by musicq** [comment]
@eawww It works fine with class component, so I think it maybe the hooks’ bug.

**Comment by fsarachu** [comment]
Looks like it's the same as #1297

**Comment by bvaughn** [comment]
React DevTools has been rewritten and [recently launched a new version 4 UI](https://reactjs.org/blog/2019/08/15/new-react-devtools.html). The source code for this rewrite was done in a separate repository and now lives in the main React repo ([github.com/facebook/react](https://github.com/facebook/react/)).

Because version 4 was a total rewrite, and all issues in this repository are related to the old version 3 of the extension, I am closing all issues in this repository. If you can still reproduce this issue, or believe this feature request is still relevant, please open a new issue in the React repo: https://github.com/facebook/react/issues/new?labels=Component:%20Developer%20Tools
